### PR TITLE
Cache number format instance

### PR DESCRIPTION
**Pull request by casperwtf**

### CODE REVIEW

File: quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java [hunk 1]
```diff
@@ -454,7 +455,11 @@ public ProxiedLocale findRelativeLanguages(@Nullable final String langCode) {
       Log.debug(Level.WARNING, "Failed to solve the player locale: " + locale + ": " + e.getMessage());
     }
 
-    return new ProxiedLocale(langCode, result, NumberFormat.getCompactNumberInstance(locale, NumberFormat.Style.SHORT), locale);
+    return new ProxiedLocale(langCode, result, getCompactNumberInstance(locale), locale);
+  }
+
+  private NumberFormat getCompactNumberInstance(@NotNull final Locale locale) {
+        return numberFormatCache.computeIfAbsent(locale, l->NumberFormat.getCompactNumberInstance(l, NumberFormat.Style.SHORT));
```
Comment (creatorfromhell) [inline quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java:464]: Should have blank line above here per the updated style guidelines.

File: quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java [hunk 2]
```diff
@@ -415,7 +416,7 @@ public ReloadResult reloadModule() {
   public ProxiedLocale findRelativeLanguages(@Nullable final String langCode) {
     //langCode may null when some plugins providing fake player
     if(langCode == null || langCode.isEmpty()) {
-      return new ProxiedLocale(langCode, DEFAULT_LOCALE, NumberFormat.getCompactNumberInstance(Locale.US, NumberFormat.Style.SHORT), Locale.ROOT);
+      return new ProxiedLocale(langCode, DEFAULT_LOCALE, getCompactNumberInstance(Locale.US), Locale.ROOT);
```
Comment (creatorfromhell) [inline quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java:420]: Should have blank line above here per the updated style guidelines.

Comment (casperwtf) [reply quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java:420]: fixed

File: quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java (hunk 1 above)
Comment (casperwtf) [reply quickshop-bukkit/src/main/java/com/ghostchu/quickshop/localization/text/SimpleTextManager.java:464]: fixed

**Review by creatorfromhell — APPROVED** [review]

**Pull request merged**